packages/pegasus/src/pegasus.js

```diff
@@ -273,287 +54,343 @@ const makePegasus = (zcf, board, namesByAddress) => {
+            transferAmountFrom(
+              amounts.Transfer,
+              'Pool',
+              poolSeat,
+              'Transfer',
+              transferSeat,
+            ),
+          transferProtocol,
+        });
+
+        const { remoteDenomToCourierPK } = localDenomState;
+
+        const courierPK = getCourierPK(remoteDenom, remoteDenomToCourierPK);
+        courierPK.resolve(courier);
+
+        const peg = makePeg(localDenomState, {
+          localBrand,
+          remoteDenom,
+          allegedName,
+        });
+        pegs.add(peg);
+        return peg;
+      },
+      abort: reason => {
+        checkAbort();
+        checkAbort = () => {
+          throw reason;
+        };
+        pegs.forEach(peg => {
+          pegToDenomState.delete(peg);
+        });
+      },
+    };
+    return Far('pegasusConnectionActions', pegasusConnectionActions);
+  };
+
   return Far('pegasus', {
-    makeDenomUri,
     /**
      * Return a handler that can be used with the Network API.
      *
-     * @returns {ConnectionHandler}
+     * @param {ERef<TransferProtocol>} [transferProtocol=DEFAULT_TRANSFER_PROTOCOL]
+     * @returns {PegasusConnectionKit}
      */
-    makePegConnectionHandler() {
+    makePegasusConnectionKit(transferProtocol = DEFAULT_TRANSFER_PROTOCOL) {
       /**
-       * @type {Store<DenomUri, Courier>}
+       * @type {WeakStore<Connection, LocalDenomState>}
        */
-      const denomUriToCourier = makeStore('Denomination');
+      // Legacy because the value contains a JS Set
+      const connectionToLocalDenomState = makeLegacyWeakMap('Connection');
+
       /**
-       * @type {Set<Peg>}
+       * @type {SubscriptionRecord<PegasusConnectionState>}
        */
-      const pegs = new Set();
-      return Far('pegConnectionHandler', {
-        async onOpen(c) {
+      const {
+        subscription: connectionSubscription,
+        publication: connectionPublication,
+      } = makeSubscriptionKit();
+
+      /** @type {ConnectionHandler} */
+      const handler = {
+        async onOpen(c, localAddr, remoteAddr) {
           // Register C with the table of Peg receivers.
-          connectionToLocalDenomState.init(c, {
-            denomUriToCourier,
-            pegs,
+          const {
+            subscription: remoteDenomSubscription,
+            publication: remoteDenomPublication,
+          } = makeSubscriptionKit();
+          const remoteDenomToCourierPK = makeLegacyMap('Denomination');
+
+          /** @type {LocalDenomState} */
+          const localDenomState = {
+            localAddr,
+            remoteAddr,
+            remoteDenomToCourierPK,
             lastDenomNonce: 0,
+            remoteDenomPublication,
+            remoteDenomSubscription,
+            abort: reason => {
+              // eslint-disable-next-line no-use-before-define
+              actions.abort(reason);
+            },
+          };
+
+          // The courier is the only thing that we use to send messages to C.
```

Comment:
```suggestion
          // The courier is the only thing that we use to send messages to `c`.
```
Is there something named upper-case `C`?